src/event_model/basemodels/event_descriptor.py

```diff
@@ -193,7 +193,7 @@ class DataKey(BaseModel):
         str, Field(description="The source (ex piece of hardware) of the data.")
     ]
     units: Annotated[
-        str,
+        Optional[str],
         Field(default="", description="Engineering units of the value"),
```

Comment:
We probably want to make the default in the basemodels `None` if that's what bluesky is doing.

```suggestion
        Field(default=None, description="Engineering units of the value"),
```